api/gen/proto/go/types/v1/extension.go

```diff
@@ -0,0 +1,6 @@
+package typesv1
+
+// IsLegacy is true if this is a legacy label names request.
+func (r *LabelNamesRequest) IsLegacy() bool {
```

Comment:
I'm not sure if this is the best place to add this.

Reply:
Done in https://github.com/grafana/pyroscope/pull/2498/commits/a10f131663ee055e670248988da4830cb707c8aa